test/KwentaNFT.OptimismKovan.test.ts

```diff
@@ -0,0 +1,224 @@
+import {GorillaBits} from '../typechain/GorillaBits'
+import recipientPrivateKeys from '../recipientPrivateKeys.json'
+
+
+describe(`KwentaNFT (Optimism Kovan)`, () => {
+  const uri = 'https://ipfs/<SAMPLE_URI>/'
+  const infuraOptimismUrl = 'https://optimism-kovan.infura.io/v3/'
+
+  let receipt: ContractReceipt,
+    KwentaNFT: KwentaNFT,
+    deployedCtc,
+    provider = new ethers.providers.JsonRpcProvider(
+      infuraOptimismUrl + process.env.INFURA_API_KEY
+    ),
+    owner: Wallet,
+    l2Wallets: Wallet[] = [],
+    _to: string[] = []
+
+  describe(`tests`, () => {
+    before(`tests`, async () => {
+      // Create list of test recipients
+      const privKeys: string[] = recipientPrivateKeys.keys
+
+      for (let i = 0; i < 206; i++) {
+        const privKey = privKeys[i]
+        const l2Wallet = new ethers.Wallet(privKey, provider)
+
+        l2Wallets.push(l2Wallet)
+      }
+
+      l2Wallets.forEach((l2Wallet: Wallet) => {
+        _to.push(l2Wallet.address)
+      })
+
+      // Create owner account
+      const ownerPrivKey = process.env.PRIVATE_KEY as string
+
+      owner = new ethers.Wallet(ownerPrivKey, provider)
+
+      const Factory__KwentaNFT = await ethers.getContractFactory('KwentaNFT')
+
+      KwentaNFT = await Factory__KwentaNFT.connect(owner).deploy(uri)
+      await KwentaNFT.deployed()
+
+      console.log(`\n KwentaNFT contract address: ${KwentaNFT.address} \n`)
+    })
+
+    /**
+     * @dev 5/5 PASS
+     */
+    describe(`uri(uint256 tokenId)`, () => {
+      it(`should get the uri for tier0 tokenIds`, async () => {
+        for (let tokenId = 1; tokenId < 101; tokenId++) {
+          const uri_ = await KwentaNFT.connect(owner).uri(tokenId)
+          expect(uri_).to.eq(uri + '0.json')
+        }
+      })
+
+      it(`should get the uri for tier1 tokenIds`, async () => {
+        for (let tokenId = 101; tokenId < 151; tokenId++) {
+          const uri_ = await KwentaNFT.connect(owner).uri(tokenId)
+          expect(uri_).to.eq(uri + '1.json')
+        }
+      })
+
+      it(`should get the uri for tier2 tokenIds`, async () => {
+        for (let tokenId = 151; tokenId < 201; tokenId++) {
+          const uri_ = await KwentaNFT.connect(owner).uri(tokenId)
+          expect(uri_).to.eq(uri + '2.json')
+        }
+      })
+
+      it(`should get the uri for tier3 tokenIds`, async () => {
+        for (let tokenId = 201; tokenId < 207; tokenId++) {
+          const uri_ = await KwentaNFT.connect(owner).uri(tokenId)
+          expect(uri_).to.eq(uri + '3.json')
+        }
+      })
+
+      it(`should throw an error for outOfRange tokenIds`, async () => {
+        for (let tokenId = 207; tokenId < 210; tokenId++) {
+          const uri_ = KwentaNFT.connect(owner).uri(tokenId)
+          await expect(uri_).to.be.revertedWith('TokenIdOutOfRange')
+        }
+      })
+    })
+
+    /**
+     * @dev 10/10 PASS
+     */
+    describe(`distribute(address[] _to)`, () => {
```

Comment:
These tests fail for me since I'm missing the file above, and my value for `_to` does not exist.